app/server/appsmith-server/src/main/java/com/appsmith/server/configurations/InstanceConfig.java

```diff
@@ -34,20 +39,64 @@ public class InstanceConfig implements ApplicationListener<ApplicationReadyEvent
 
     private boolean isRtsAccessible = false;
 
+    @Autowired
+    private ApplicationContext applicationContext;
+
     @Override
     public void onApplicationEvent(ApplicationReadyEvent applicationReadyEvent) {
-        configService.getByName(Appsmith.APPSMITH_REGISTERED)
+
+        Mono<Void> registrationAndRtsCheckMono = configService.getByName(Appsmith.APPSMITH_REGISTERED)
                 .filter(config -> Boolean.TRUE.equals(config.getConfig().get("value")))
                 .switchIfEmpty(registerInstance())
                 .doOnError(errorSignal -> log.debug("Instance registration failed with error: \n{}", errorSignal.getMessage()))
                 .then(performRtsHealthCheck())
-                .doFinally(ignored -> this.printReady())
+                .doFinally(ignored -> this.printReady());
+
+        Mono.zip(checkInstanceSchemaVersion(), registrationAndRtsCheckMono)
```

Comment:
Since we are expecting empty, and don't care about the zipped results of these `Mono`s, what do you think about using `Mono.when` instead? `Mono.when` is explicitly for the purpose of just waiting, and then resolving to a `Mono.empty`, exactly what we need here.

```suggestion
        Mono.when(checkInstanceSchemaVersion(), registrationAndRtsCheckMono)
```